src/py42/services/audit_logs.py

```diff
@@ -0,0 +1,63 @@
+from py42 import settings
+from py42.services import BaseService
+from py42.services.util import get_all_pages
+from py42.util import convert_datetime_to_timestamp_str
+
+
+class AuditLogsService(BaseService):
+    def get_page(self, page_num, page_size=None, **kwargs):
+
+        uri = u"/rpc/search/search-audit-log"
+        page_size = page_size or settings.items_per_page
+        params = dict(
+            page=page_num - 1,
+            pageSize=page_size,
+            dateRange=kwargs["date_range"],
+            eventTypes=kwargs["event_types"],
+            actorIds=kwargs["user_ids"],
+            actorNames=kwargs["user_names"],
+            actorIpAddresses=kwargs["user_ip_addresses"],
+            affectedUserIds=kwargs["affected_user_ids"],
+            affectedUserNames=kwargs["affected_user_names"],
+        )
```

Comment:
I anticipate that as we (as a company) add functionality to the audit log system, this set of available params will expand. We should maybe design this so that if additional unknown kwargs are passed, they're just added to the params, allowing it to be possible to use new params before py42 is updated. 

Something like: 

```
date_range = kwargs.pop("date_range")
event_types = kwargs.pop("event_types")
... etc

params = dict(
            page=page_num - 1,
            pageSize=page_size,
            dateRange=date_range,
            eventTypes=event_types,
            ....etc
            **kwargs)
```

Reply:
It depends on how API behaves, as this particular API does so right now, if API fails if even one parameter/field name is incorrect then the suggested approach will fail when incorrect param names are passed, e.g we need to convert `event_types` to `eventTypes` such names cannot be anticipated advance and we would end up passing 'event_types'.

Having said that this is a good point, since we are building API gateway @alanag13 can make a note. The contract of every API should be that adding new optional parameters in future should not break/fail when those new fields/params are not passed. That way at-least we will save the existing implementation from breaking with new releases.

Reply:
If the API fails because the user put something weird in `**kwargs`, I think that is on them.  Having `kwargs` allows more people to have support for the time being and to be honest, I think every API method could have them.

Reply:
So does it mean `get_all` method should accept `**kwargs`?

Reply:
Added.

Reply:
`sdk.auditlogs.get_all(deviceUserName='xyz')` will return all results, hope that is expected.

Reply:
The main goal here is making our own lives easier if the api gets updated. This will allow us to "patch" any integrations using an older py42 with the new values rather than forcing a full update of py42 (which could take awhile to get released, could have other changes that make upgrading more work, etc.). 

End-users don't really need to know why kwargs are accepted, but we _should_ have all currently known args spelled out as optional args, to make it easy for them to use (we don't want them to have to always look to the docs to get possible arg values). 

That said, we _do_ need to be a bit careful on how we process these. For example:

An integration consuming audit logs is using the bare function call to get all results:

```
results = sdk.auditlogs.get_all()
```

The auditlogs service is updated to include new event types, and adds a new param option: `affectedOrgs`. Unfortunately they made it a required param, so our integration is currently broken. 

We can patch the integration by just adding the missing param to the extra kwargs:

```
results = sdk.auditlogs.get_all(affectedOrgs=None)
```

However, on the next py42 update, if we add an official `affected_orgs` param but our downstream integration doesn't change the param name when updating py42, it could break, because then the params dict:

```
params = dict(
            page=page_num - 1,
            pageSize=page_size,
            dateRange=date_range,
            eventTypes=event_types,
            actorIds=user_ids,
            actorNames=usernames,
            actorIpAddresses=user_ip_addresses,
            affectedUserIds=affected_user_ids,
            affectedUserNames=affected_usernames,
            affectedOrgs=affected_orgs,
            **kwargs
        )
``` 
will now be given two instances of `affectedOrgs` args. 

So to be safe we should instead of passing the final `**kwargs` into the dict() call, we should construct the dict with all of our known params, then update it with the `**kwargs`:

```
params = dict(
            page=page_num - 1,
            pageSize=page_size,
            dateRange=date_range,
            eventTypes=event_types,
            actorIds=user_ids,
            actorNames=usernames,
            actorIpAddresses=user_ip_addresses,
            affectedUserIds=affected_user_ids,
            affectedUserNames=affected_usernames,
            affectedOrgs=affected_orgs
)
params.update(kwargs)
```

That way we _can_ patch if needed, but also not break the patched code once py42 is upgraded later on.

Reply:
I'm fine with adding `**kwargs`, but we should keep the known arguments as named. This is mainly to provide a workaround in the event that an api feature gets introduced before we add an accompanying named parameter, and if the user uses parameter names that don't exist, that will be on them.

Reply:
First we would do the patch work and then the proper work kind of seems weird, changed nevertheless.